src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/SessionService.kt

```diff
@@ -246,4 +248,26 @@ class SessionService(
 
   private fun isDateWithinRange(sessionDate: LocalDate, startDate: LocalDate, endDate: LocalDate? = null) =
     sessionDate >= startDate && (endDate == null || sessionDate <= endDate)
+
+  fun getSessionCapacity(prisonCode: String, sessionDate: LocalDate, sessionStartTime: LocalTime, sessionEndTime: LocalTime): SessionCapacityDto {
+
+    val dayOfWeek = sessionDate.dayOfWeek
+
+    val sessionTemplates = sessionTemplateRepository.findValidSessionTemplatesForSession(
+      prisonCode,
+      sessionDate,
+      sessionStartTime,
+      sessionEndTime,
+      dayOfWeek
+    )
+
+    if (sessionTemplates.isEmpty()) {
+      throw CapacityNotFoundException("Session capacity not found prisonCode:$prisonCode,session Date:$sessionDate, StartTime:$sessionStartTime, EndTime:$sessionEndTime, dayOfWeek:$dayOfWeek")
+    }
+    if (sessionTemplates.size > 1) {
```

Comment:
In case of bi-weekly sessions we might get multiple results back.

Reply:
true